changelog/3016.bugfix

```diff
@@ -0,0 +1,7 @@
+internal pytest test suite improvements:
```

Comment:
I don't think we can have sub-items in `newsfiles` like this because it will break up the RST lint checker. You can run `inv changelog` to update the changelog and check it yourself.

Reply:
Took a while to figure out what `inv changelog` meant or how to get it working for the project. You really should add a tox environment for this and add a note in the contributing docs on how to run all these processes.

needed to:
- figure out where the change log is generated from
- what `invoke` was
- what I need to have installed to run it
- what arguments to pass to invoke

in the end, the best way seems to have been:
- create an environment
- install the packages from tasks/requirements.txt
- run `towncrier --draft` to see the generated changelog in the console output without anything else getting modified in the project